homeassistant/components/wemo/sensor.py

```diff
@@ -87,36 +86,27 @@ def _update(self, force_update=True) -> None:
 class InsightCurrentPower(InsightSensor):
     """Current instantaineous power consumption."""
 
-    def __init__(self, device: DeviceWrapper, update_insight_params: Callable) -> None:
-        """Initialize the WeMo Insight power sensor."""
-        super().__init__(
-            device,
-            update_insight_params,
-            "Current Power",
-            DEVICE_CLASS_POWER,
-            POWER_WATT,
-        )
+    _attr_device_class = DEVICE_CLASS_POWER
+    _attr_unit_of_measurement = POWER_WATT
+    _name_suffix = "Current Power"
+    _insight_params_key = "currentpower"
```

Comment:
We actually just added a generic approach for this 7 hours ago :) #53357

Reply:
A refactor using the new EntityDescription would be a nice follow up PR.

Reply:
#53537 for the follow up.